lib/p5.play.js

```diff
@@ -1480,72 +1422,80 @@ function Sprite(pInst, _x, _y, _w, _h) {
+  * parameter passed at the creation of the sprite or the from the image
+  * dimension in case of animated sprites.
   *
   * Often the image bounding box is not appropriate as the active area for
   * collision detection so you can set a circular or rectangular sprite with
   * different dimensions and offset from the sprite's center.
   *
-  * There are four ways to call this method:
+  * There are many ways to call this method.  The first argument determines the
+  * type of collider you are creating, which in turn changes the remaining
+  * arguments.  Valid collider types are:
+  *
+  * * `point` - A point collider with no dimensions, only a position.
+  *
+  *   `setCollider("point"[, offsetX, offsetY])`
+  *
+  * * `circle` - A circular collider with a set radius.
+  *
+  *   `setCollider("circle"[, offsetX, offsetY[, radius])`
+  *
+  * * `rectangle` - An alias for `aabb`, below.
+  *
+  * * `aabb` - An axis-aligned bounding box - has width and height but no rotation.
+  *
+  *   `setCollider("aabb"[, offsetX, offsetY[, width, height]])`
+  *
+  * * `obb` - An oriented bounding box - has width, height, and rotation.
+  *
+  *   `setCollider("obb"[, offsetX, offsetY[, width, height[, rotation]]])`
   *
-  * 1. setCollider("rectangle")
-  * 2. setCollider("rectangle", offsetX, offsetY, width, height)
-  * 3. setCollider("circle")
-  * 4. setCollider("circle", offsetX, offsetY, radius)
   *
   * @method setCollider
-  * @param {String} type Either "rectangle" or "circle"
-  * @param {Number} offsetX Collider x position from the center of the sprite
-  * @param {Number} offsetY Collider y position from the center of the sprite
-  * @param {Number} width Collider width or radius
-  * @param {Number} height Collider height
+  * @param {String} type One of "point", "circle", "rectangle", "aabb" or "obb"
+  * @param {Number} [offsetX] Collider x position from the center of the sprite
+  * @param {Number} [offsetY] Collider y position from the center of the sprite
+  * @param {Number} [width] Collider width or radius
+  * @param {Number} [height] Collider height
+  * @param {Number} [rotation] Collider rotation in degrees
   * @throws {TypeError} if given invalid parameters.
   */
-  this.setCollider = function(type, offsetX, offsetY, width, height) {
-    if (!(type === 'rectangle' || type === 'circle')) {
-      throw new TypeError('setCollider expects the first argument to be either "circle" or "rectangle"');
-    } else if (type === 'circle' && !(arguments.length === 1 || arguments.length === 4)) {
-      throw new TypeError('Usage: setCollider("circle") or setCollider("circle", offsetX, offsetY, radius)');
-    } else if (type === 'rectangle' && !(arguments.length === 1 || arguments.length === 5)) {
-      throw new TypeError('Usage: setCollider("rectangle") or setCollider("rectangle", offsetX, offsetY, width, height)');
+  this.setCollider = function(type, offsetX, offsetY, width, height, rotation) {
```

Comment:
`setCollider`'s a weird one - there are literally more than a dozen correct ways to call it, and a lot more incorrect ways.  Hence the overzealous validation (and later, tests over said validation).